### PR TITLE
Accept a secret with TLS certs

### LINKED ISSUE #89: ImageAutomation: Custom TLS certs for container registries
For the internal Docker registry we use a TLS certificate to authenticate, instead of username/password.
The image update automation in FluxV2 lacks support for using a custom TLS certificate to authenticate towards docker registry.
In FluxV1, we used the `registry.insecureHosts` helm property as a work-around.

`flux get image repository`
```
NAME            READY   MESSAGE                                                                                 LAST SCAN       SUSPENDED
<REDACTED>        False   Get "<docker-registry-url>": x509: certificate signed by unknown authority                       False
```

`flux --version`
flux version 0.5.4

### PR DESCRIPTION
**Pull request by squaremo**
This implements the design in #89, so you can supply a client cert and key to use for authentication; and, separately, a CA certificate if you are using self-signed certificates.

Fixes #89.

### CODE REVIEW

File: api/v1alpha1/imagerepository_types.go [hunk 1]
```diff
@@ -49,6 +49,20 @@ type ImageRepositorySpec struct {
...
+	//  - a PEM-encoded client certificate (`certFile`) and private
+	//  key (`keyFile`);
+	//  - a PEM-encoded CA certificate (`cacertFile`)
+	//
+	//  and whichever are supplied, will be used for connecting to the
+	//  registry. The client cert and key are useful if you are
+	//  authenticating with a certificate; the CA cert is useful if
+	//  you are using a self-signed server certificate.
+	// +optional
+	CertSecretRef *corev1.LocalObjectReference `json:"certSecretRef,omitempty"`
```
Comment (stefanprodan) [inline api/v1alpha1/imagerepository_types.go:64]: Please use `LocalObjectReference ` from `fluxcd/pkg/apis/meta` package.

Comment (stefanprodan) [reply api/v1alpha1/imagerepository_types.go:64]: You need to rebase first to get https://github.com/fluxcd/image-reflector-controller/commit/539409d3db591d6e503952425362968e61518999

Comment (squaremo) [reply api/v1alpha1/imagerepository_types.go:64]: Doing so seems to break code generation :-(

```
$ make generate manifests api-docs 
go: creating new go.mod: module tmp
go: found sigs.k8s.io/controller-tools/cmd/controller-gen in sigs.k8s.io/controller-tools v0.3.0
cd api; /Users/mikeb/go/bin/controller-gen object:headerFile="../hack/boilerplate.go.txt" paths="./..."
230:20: missing ',' in argument list
235:20: missing ',' in argument list
Error: not all generators ran successfully
run `controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./... -w` to see all available markers, or `controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./... -h` for usage
make: *** [generate] Error 1
```

Comment (squaremo) [reply api/v1alpha1/imagerepository_types.go:64]: Are they used anywhere else?

Comment (squaremo) [reply api/v1alpha1/imagerepository_types.go:64]: nvm -- updating controller-gen gave me better errors (better than line numbers without a file!), making it obvious that I needed to `(cd api; go get github.com/fluxcd/apis/meta)`.

File: controllers/imagerepository_controller.go [hunk 2]
```diff
@@ -46,6 +49,12 @@ import (
 	imagev1alpha1 "github.com/fluxcd/image-reflector-controller/api/v1alpha1"
 )
 
+const (
+	ClientCert = "certFile"
+	ClientKey  = "keyFile"
+	CACert     = "cacertFile"
```
Comment (stefanprodan) [inline controllers/imagerepository_controller.go:55]:
```suggestion
	CACert     = "caFile"
```

To match Helm https://github.com/fluxcd/flux2/blob/main/cmd/flux/create_secret_helm.go#L122

**Review by stefanprodan — CHANGES_REQUESTED** [review]

File: controllers/imagerepository_controller.go (hunk 2 above)
Comment (squaremo) [reply controllers/imagerepository_controller.go:55]: Huh, I looked this up in source-controller/internal/helm and somehow transcribed it wrong 😳

**Review by stefanprodan — COMMENTED** [review]
Can you please update the spec docs and add examples on how to create a secret with kubectl?

**Comment by squaremo** [comment]
> update the spec docs and add examples on how to create a secret with kubectl

Good shout; yep.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @squaremo 🍻

**Pull request merged**